### PR TITLE
HDDS-4035. Update logs of HadoopDirGenerator.

### PR DESCRIPTION
**Pull request by aryangupta1998**
## What changes were proposed in this pull request?

1. Added logs to HadoopNestedDirGenerator.

2. Edit logs of HadoopDirTreeGenerator. 

3. Use BaseFreonGenerator printReport() for shell and command-line execution of freon commands like ddsg, dtsg, etc.
## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-4035

## How was this patch tested?

Tested Manually.

### CODE REVIEW

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/HadoopNestedDirGenerator.java [hunk 1]
```diff
@@ -109,5 +117,7 @@ private void createDir(long counter) throws Exception {
       Path dir = new Path(rootPath.concat("/").concat(childDir));
       fileSystem.mkdirs(dir.getParent());
     }
+    System.out.println("\nSuccessfully created directories. Total " +
```
Comment (rakeshadr) [inline hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/HadoopNestedDirGenerator.java:120]: Sometimes freon is executed from 'shell' sometimes from 'command line', BaseFreonGenerator.printReport already handle this and we can follow this approach. 

Along with this patch, can you change the "System.out" in https://github.com/apache/hadoop-ozone/blob/master/hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/HadoopDirTreeGenerator.java#L154 as well. Thanks!

Comment (aryangupta1998) [reply hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/HadoopNestedDirGenerator.java]: Thanks for the review Rakesh. I have updated the PR and also the logs of HadoopTreeDirGenerator as per your suggestions.

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/BaseFreonGenerator.java [hunk 2]
```diff
@@ -301,6 +301,15 @@ public void printReport() {
     messages.forEach(print);
   }
 
+  public void print(String s){
```
Comment (rakeshadr) [inline hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/BaseFreonGenerator.java:304]: Latest patch looks good. Adding few minor comments:

Please add javadoc and change the parameter name 's' to 'msg', just to improve more readable.

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/HadoopDirTreeGenerator.java [hunk 3]
```diff
@@ -97,17 +97,25 @@
...
+    } else {
+      init();
+      OzoneConfiguration configuration = createOzoneConfiguration();
+      fileSystem = FileSystem.get(URI.create(rootPath), configuration);
+
+      contentGenerator = new ContentGenerator(fileSizeInBytes, bufferSize);
+      timer = getMetrics().timer("file-create");
+
+      runTests(this::createDir);
+      return null;
```
Comment (rakeshadr) [inline hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/HadoopDirTreeGenerator.java:116]: minor comment: please remove 'return null' inside the else block as below return is sufficient.

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/BaseFreonGenerator.java (hunk 2 above)
Comment (aryangupta1998) [reply hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/BaseFreonGenerator.java:304]: Added javadoc, changed 's' to 'msg'.

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/HadoopDirTreeGenerator.java (hunk 3 above)
Comment (aryangupta1998) [reply hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/HadoopDirTreeGenerator.java:116]: Removed 'return null' statement inside else block.

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/BaseFreonGenerator.java [hunk 4]
```diff
@@ -301,6 +301,19 @@ public void printReport() {
     messages.forEach(print);
   }
 
+  /**
+   * Print out the messages for HadoopNestedDirGenerator
```
Comment (rakeshadr) [inline hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/BaseFreonGenerator.java:305]: As this is base class, please remove the inherited class details here.

you can write something like,
"Print out reports with the given message."

Comment (aryangupta1998) [reply hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/BaseFreonGenerator.java]: Changed the javadoc.

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/BaseFreonGenerator.java [hunk 5]
```diff
@@ -301,6 +301,18 @@ public void printReport() {
     messages.forEach(print);
   }
 
+  /**
+   * Print out reports with the given message.
+   */
+  public void print(String msg){
+    List<String> messages = new LinkedList<>();
+    messages.add(msg);
```
Comment (xiaoyuyao) [inline hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/BaseFreonGenerator.java:309]: Given we only have one String msg here, we can eliminate thew List allocation here by just simply call
print.accept(msg) in line 313.

Comment (aryangupta1998) [reply hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/freon/BaseFreonGenerator.java]: Thanks @xiaoyuyao. Fixed the print function.

**Comment by rakeshadr** [comment]
@aryangupta1998, Thanks for updating the PR!

The changes look good to me. Pending GREEN CI

**Comment by rakeshadr** [comment]
Merged the changes to master branch. Thanks @aryangupta1998 for the contribution!

**Pull request merged**